sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumerClient/EventHubConsumerClientTests.cs

```diff
@@ -550,16 +442,61 @@ public async Task CloseAsyncClosesTheTransportConsumer()
         /// </summary>
         ///
         [Test]
-        public void CloseClosesTheTransportConsumer()
+        public async Task CloseAsyncSurfacesExceptionsForActiveTransportConsumers()
+        {
+            var mockTransportConsumer = new Mock<TransportConsumer>();
+            var mockConnection = new MockConnection(() => mockTransportConsumer.Object);
+            var consumer = new EventHubConsumerClient(EventHubConsumerClient.DefaultConsumerGroupName, mockConnection);
+
+            mockTransportConsumer
+                .Setup(consumer => consumer.CloseAsync(It.IsAny<CancellationToken>()))
+                .Returns(Task.FromException(new InvalidCastException()));
+
+            try { await using var receiver = consumer.CreatePartitionReceiver("0", EventPosition.Earliest); } catch {}
+
+            Assert.That(async () => await consumer.CloseAsync(), Throws.InstanceOf<Exception>());
```

Comment:
Couldn't we expect an `InvalidCastException`?

Reply:
It seems that's what we do in the sync `Close` test.

Reply:
Good catch.   Forgot to fix that after I figured out that I needed that `try/catch` block.  😄